src/components/node-list/node-list-tree.js

```diff
@@ -0,0 +1,179 @@
+    }
+  };
+
+  const renderTree = (
+    rowData,
+    onItemMouseEnter,
+    onItemMouseLeave,
+    onItemChange,
+    onItemClick,
+    searchTreeCount
+  ) => (
+    <NodeListTreeItem
+      data={rowData}
+      onItemMouseEnter={onItemMouseEnter}
+      onItemMouseLeave={onItemMouseLeave}
+      onItemChange={onItemChange}
+      onItemClick={onItemClick}
+      key={rowData.id}
+      searchCount={searchTreeCount[rowData.id]}>
+      {rowData.children.length > 0 &&
+        rowData.children.map((node) =>
+          renderTree(
+            node,
+            onItemMouseEnter,
+            onItemMouseLeave,
+            onItemChange,
+            onItemClick,
+            searchTreeCount
+          )
+        )}
+
+      {/* render set of node elements in that modular pipeline */}
+      {rowData.nodes.map((node) => (
+        <NodeListTreeItem
+          data={node}
+          onItemMouseEnter={onItemMouseEnter}
+          onItemMouseLeave={onItemMouseLeave}
+          onItemChange={onItemChange}
+          onItemClick={onItemClick}
+          key={node.id}
+        />
+      ))}
+    </NodeListTreeItem>
+  );
+
+  return (
+    <StyledTreeView
+      className={classes.root}
+      defaultCollapseIcon={<ExpandMoreIcon />}
+      defaultExpandIcon={<ChevronRightIcon />}>
+      {treeData.children.length > 0 &&
+        treeData.children.map((node) =>
+          renderTree(
+            node,
+            onItemMouseEnter,
+            onItemMouseLeave,
+            onItemChange,
+            onItemClick,
+            searchTreeCount
+          )
+        )}
+
+      {/* render set of node elements in the main pipeline */}
+      {treeData.nodes.map((node) => (
+        <NodeListTreeItem
+          data={node}
+          onItemMouseEnter={onItemMouseEnter}
+          onItemMouseLeave={onItemMouseLeave}
+          onItemChange={onItemChange}
+          onItemClick={onItemClick}
+          key={node.id}
+        />
+      ))}
+    </StyledTreeView>
+  );
+};
+
+export const mapStateToProps = (state) => ({
+  nodes: getGroupedNodes(state),
+  nodeSelected: getNodeSelected(state),
+  nodeModularPipelines: getNodeModularPipelines(state),
+  types: getNodeTypes(state),
+  nodeTypeIDs: getNodeTypeIDs(state),
+  modularPipelineIds: getModularPipelineIDs(state),
+  modularPipelines: getModularPipelineData(state),
+});
+
+export const mapDispatchToProps = (dispatch) => ({
+  onToggleModularPipelineActive: (modularPipelineIDs, active) => {
+    dispatch(toggleModularPipelineActive(modularPipelineIDs, active));
+  },
+  onToggleModularPipelineFilter: (modularPipelineIDs, enabled) => {
+    dispatch(toggleModularPipelineFilter(modularPipelineIDs, enabled));
+  },
+  onToggleTypeDisabled: (typeID, disabled) => {
+    dispatch(toggleTypeDisabled(typeID, disabled));
+  },
+  onToggleNodeSelected: (nodeID) => {
+    dispatch(loadNodeData(nodeID));
+  },
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(TreeListProvider);
```

Comment:
I am a little confused there are 2 TreeListProviders -- one in node-list-tree.js and one in node-list-tree-search.js. Is there code that needs to be removed?

Reply:
I'm glad you raised this Rashida - the main reason for setting a new component for search is given the need to remount the `treeList` component to enable the search UI with the expanded tree ( otherwise it would generate a react error in regards to changing a controlled component to an uncontrolled element due to the unique setup of material-ui), and indeed it does causes repetitive code. 

I had separated the search-ui tree out as a separate component to provide a clear seperation of the search-ui logic away from the normal tree-ui (instead of incorporating it in the existing tree-list provider) so that I could easily strip that away once I find the solution to fix the issue with the react warning I mentioned above, which is a temporary solution hence the repetitive code. I should have added a note to explain that intention.

That said, i agree with what you mentioned above, and had refactored and incorporated the search logic into the original tree-list provider.